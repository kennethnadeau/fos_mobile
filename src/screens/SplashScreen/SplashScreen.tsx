import React, {useEffect} from 'react';
import {StyleSheet} from 'react-native';
import {ScreenFC} from 'react-native-navigation-register-screens';
import Logo from '@fos/components/Logo';
import {SCREENS} from '@fos/constants';
import {SafeAreaView} from 'react-native-safe-area-context';
import {vs} from 'react-native-size-matters';
import {Navigation} from 'react-native-navigation';

const SplashScreen: ScreenFC = () => {
  // TODO Remove when delay when startup logic is ready
  useEffect(() => {
    const timeoutId = setTimeout(
      () =>
        Navigation.setRoot({
          root: {
            component: {
              name: SCREENS.NAV_UX,
            },
          },
        }),
      800,
    );

    return () => clearTimeout(timeoutId);
  }, []);

  return (
    <SafeAreaView style={styles.container}>
      <Logo />
    </SafeAreaView>
  );
};

SplashScreen.screenName = SCREENS.SPLASH_SCREEN;

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'space-around',
    marginTop: vs(20),
  },
});

export default SplashScreen;
