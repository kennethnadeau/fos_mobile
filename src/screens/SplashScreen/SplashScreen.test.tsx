import React from 'react';
import SplashScreen from './SplashScreen';
import {render, RenderAPI} from 'react-native-testing-library';

describe('<SplashScreen />', () => {
  let rendered: RenderAPI;

  beforeEach(() => {
    rendered = render(<SplashScreen componentId="id" />);
  });

  it('should show logo at the top', () => {
    const actual = rendered.getByA11yLabel('App logo');

    expect(actual).toBeDefined();
  });
  it('should match snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
