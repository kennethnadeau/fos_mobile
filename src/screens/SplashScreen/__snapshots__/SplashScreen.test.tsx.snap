// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplashScreen /> should match snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
      "marginTop": 39.23529411764706,
    }
  }
>
  <Image
    accessibilityLabel="App logo"
    accessible={true}
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/images/logo.png",
      }
    }
    style={
      Array [
        Object {
          "height": 270.7235294117647,
          "width": 323.57142857142856,
        },
        undefined,
      ]
    }
  />
</RNCSafeAreaView>
`;
